tests/func/test_repro.py

```diff
@@ -1611,30 +1611,46 @@ def test(self):
         assert evaluation[2].relpath == "E.dvc"
 
 
-def test_ssh_dir_out(tmp_dir, dvc):
-    if not _should_test_ssh():
-        pytest.skip()
-
+def test_ssh_dir_out(tmp_dir, dvc, ssh_server):
     tmp_dir.gen({"foo": "foo content"})
 
     # Set up remote and cache
-    remote_url = get_ssh_url()
+    user = ssh_server.test_creds["username"]
+    port = ssh_server.port
+    keyfile = ssh_server.test_creds["key_filename"]
+
+    remote_url = get_ssh_url_mocked(user, port)
     assert main(["remote", "add", "upstream", remote_url]) == 0
+    assert main(["remote", "modify", "upstream", "keyfile", keyfile]) == 0
 
-    cache_url = get_ssh_url()
+    cache_url = get_ssh_url_mocked(user, port)
     assert main(["remote", "add", "sshcache", cache_url]) == 0
     assert main(["config", "cache.ssh", "sshcache"]) == 0
+    assert main(["remote", "modify", "sshcache", "keyfile", keyfile]) == 0
 
     # Recreating to reread configs
     repo = DvcRepo(dvc.root_dir)
 
+    # To avoid "WARNING: UNPROTECTED PRIVATE KEY FILE" from ssh
+    os.chmod(keyfile, 0o600)
+
+    (tmp_dir / "script.py").write_text(
+        "import sys, pathlib\n"
+        "path = pathlib.Path(sys.argv[1])\n"
+        "dir_out = path / 'dir-out'\n"
+        "dir_out.mkdir()\n"
+        "(dir_out / '1.txt').write_text('1')\n"
+        "(dir_out / '2.txt').write_text('2')\n"
+    )
     url_info = URLInfo(remote_url)
-    mkdir_cmd = "mkdir dir-out;cd dir-out;echo 1 > 1.txt; echo 2 > 2.txt"
```

Comment:
This actually didn't work because of the bug in `remove`. That is why you need to use `&&` to ensure that no error occurs. Same complaint as when people don't use `set -e \n set -x` in their bash scripts 🙂